agent/engine/docker_task_engine.go

```diff
@@ -830,6 +832,15 @@ func (engine *DockerTaskEngine) Capabilities() []string {
 		}
 	}
 
+	if engine.cfg.TaskResourceLimitEnabled {
```

Comment:
Can you add a TODO here talking about ensuring compatibility with existing, already-running tasks (upgrade scenario)?  If there are already tasks running without a cgroup parent, we need to disable this feature.

Reply:
I'm tracking that separately; I will also add a TODO